trino-rest-github/src/main/java/pl/net/was/rest/github/GithubRest.java

```diff
@@ -2127,4 +2196,89 @@ public ConnectorSplitSource getSplitSource(
         }
         return new FixedSplitSource(splits.build());
     }
+
+    @Override
+    public TableStatistics getTableStatistics(
+            ConnectorSession session,
+            ConnectorTableHandle handle,
+            Constraint constraint)
+    {
+        RestTableHandle table = (RestTableHandle) handle;
+        GithubTable tableName = GithubTable.valueOf(table);
+        Map<String, ColumnHandle> columns = columnHandles.get(tableName);
+        FilterApplier filter = filterAppliers.get(tableName);
+        TableStatistics.Builder builder = TableStatistics.builder();
+
+        Function<RestTableHandle, Long> getter = rowCountGetters.get(tableName);
+        int multiplier = 1;
+        if (tableName == GithubTable.CHECK_RUN_ANNOTATIONS) {
+            getter = rowCountGetters.get(GithubTable.CHECK_RUNS);
+            multiplier = 4;
+        }
+        else if (tableName == GithubTable.JOBS) {
+            // because jobs require run_id, which would be provided after applying a dynamic filter from the join
+            // fake a huge number here to ensure this is always the right side of the join
+            getter = rowCountGetters.get(GithubTable.RUNS);
+            multiplier = 100;
+        }
+        if (getter != null) {
+            // constraint needs to be pushed down into the table handle
+            if (filter != null) {
+                Optional<ConstraintApplicationResult<ConnectorTableHandle>> result = filter.applyFilter(
+                        table,
+                        columns,
+                        filter.getSupportedFilters(),
+                        constraint.getSummary());
+                if (result.isPresent()) {
+                    table = (RestTableHandle) result.get().getHandle();
+                }
+            }
+            long totalCount = multiplier * getter.apply(table);
```

Comment:
If this complaints about missing conditions, ignore it.

Also, make sure that the raised exception includes the table name.